db/migrate/20210312191925_add_username_tsvector_index_to_users.rb

```diff
@@ -0,0 +1,24 @@
+class AddUsernameTsvectorIndexToUsers < ActiveRecord::Migration[6.0]
+  disable_ddl_transaction!
+
+  INDEX = "to_tsvector('simple'::regconfig, COALESCE((username)::text, ''::text))"
+  private_constant :INDEX
+
+  def up
+    return if index_exists?(:users, INDEX, name: "index_usernames_as_tsvector")
```

Comment:
Only a non blocking tiny detail: the index naming structure in Rails is `index_{TABLE}_on_{FIELD}`, so if we want to follow a similar pattern, we could use `index_users_on_username_as_tsvector`